### DISCUSSION TITLE
Textures in DirectX 11

### DISCUSSION THREAD

**Discussion by jjb-pro** [dotnet/Silk.NET#1845] [Q&A] [answered]
Hello, 
I'm currently working on a small project with Silk.NET and DirectX 11. Everything is working really well so far. However, I am now trying to load textures. Shader, SamplerState, etc. are already set up. 

How can I load a texture (all common formats _*.dds_, _*.tga_, _*.png_, _*.jpg_, etc.) in C# as `ID3D11ShaderResourceView`? 

In C++, you would add Windows Imaging Component (WIC) and then load the texture with the following line: 
```cpp
DirectX::CreateWICTextureFromFile(device, L"path.jpg", &Resource, &ShaderResourceView);
```

Thanks for the help!

**Comment by curin** [comment, marked as answer]
Since we don't have bindings to the directx XTK that function you are mentioning isn't available, but you could also import that function yourself if you really wanted to. Otherwise, this is some older code I wrote for DX12 (uses sharpdx from when that was still active), but the code is pretty much identical to how you would do this in silk. https://github.com/curin/CPU-GPU-Renderer/blob/master/GPUShaders/ShaderProfiles/SimpleTexture.cs#L195. It loads a bitmap as C# normally would and then converts it to a byte buffer with proper ordering (depends on what your resource is setup as rgba, rgb, argb, etc.) Then it writes it to a resource. This is essentially what the XTK function does for you. The DX11 code for this will be a little different, but have similar steps.

**Reply by jjb-pro** [reply]
Thank you! I have written my own C++ wrapper and can use it via the DllImportAttribute in C#. It works great with Silk.NET.
